### PR TITLE
feat(1455): Allow create or update pipeline with rootDir (5)

### PR DESCRIPTION
**Pull request by tkyi**
## Context
Would be nice if users could put screwdriver.yaml/source dir not at root.

## Objective
This PR adds rootDir to pipeline create and update. Also adds helper function to make sure rootDir does not start with or end with a `/`.

## Related links
Related to https://github.com/screwdriver-cd/screwdriver/issues/1455

### CODE REVIEW

File: plugins/pipelines/helper.js [hunk 1]
```diff
@@ -26,6 +26,27 @@ const formatCheckoutUrl = (checkoutUrl) => {
...
+/**
+ * Get rid of leading/trailing slashes in rootDir, return empty string as default
+ * @method sanitizeRootDir
+ * @param  {String}     rootDir     Root directory (ex: /src/component/app/ or /)
+ * @return {String}                 Root dir with no leading/trailing slashes
+ */
+const sanitizeRootDir = (rootDir = '') => {
+    let result = rootDir;
+
+    if (rootDir.startsWith('/')) {
```
Comment (d2lam) [inline plugins/pipelines/helper.js]: shouldn't we just get rid of all extra `/`. 
For example: `//src` --> `src`

File: plugins/pipelines/helper.js [hunk 2]
```diff
@@ -26,6 +26,27 @@ const formatCheckoutUrl = (checkoutUrl) => {
...
+        result = result.slice(1);
```
Comment (d2lam) [inline plugins/pipelines/helper.js]: shouldn't we just get rid of all extra /.
For example: //src --> src

File: plugins/pipelines/helper.js [hunk 3]
```diff
@@ -26,6 +26,27 @@ const formatCheckoutUrl = (checkoutUrl) => {
...
+    }
+
+    if (rootDir.endsWith('/')) {
+        result = result.slice(0, -1);
```
Comment (d2lam) [inline plugins/pipelines/helper.js]: same comment as above

File: plugins/pipelines/helper.js (hunk 2 above)
Comment (tkyi) [reply plugins/pipelines/helper.js]: I think that's fair
updated method

File: plugins/pipelines/helper.js [hunk 4]
```diff
@@ -26,6 +26,15 @@ const formatCheckoutUrl = (checkoutUrl) => {
     return result;
 };
 
+/**
+ * Get rid of leading/trailing slashes in rootDir, return empty string as default
+ * @method sanitizeRootDir
+ * @param  {String}     rootDir     Root directory (ex: /src/component/app/ or /)
+ * @return {String}                 Root dir with no leading/trailing slashes
+ */
+const sanitizeRootDir = (rootDir = '') => rootDir.replace(/^\/+|\/+$/g, '');
```
Comment (jithin1987) [inline plugins/pipelines/helper.js]: can we also validate if it's an actual path ?

Comment (minz1027) [reply plugins/pipelines/helper.js]: you mean check if it exists in the repo? that's going to be hard, need to make git calls

Comment (jithin1987) [reply plugins/pipelines/helper.js]: Not checking if it actually exits in repo, just patterns for valid file path. Eg: /x/y/z or ./s or s/y

Comment (tkyi) [reply plugins/pipelines/helper.js]: I think that would be hard to do correctly. Here are some of the top results based on a quick search online:
- https://stackoverflow.com/questions/16231210/file-path-validation-in-javascript
- https://www.codeproject.com/Tips/216238/Regular-Expression-to-Validate-File-Path-and-Exten
- https://www.npmjs.com/package/is-valid-path (this package was last published 4 years ago)

If you really want it, I can try to adapt one of those regex patterns.

Comment (jithin1987) [reply plugins/pipelines/helper.js]: we need to look into only valid unix file paths.  

https://stackoverflow.com/questions/28989672/regex-to-tell-if-a-string-contains-a-linux-file-path-or-if-a-linux-file-path-as
https://stackoverflow.com/questions/55069650/regex-pattern-to-validate-linux-folder-path

Comment (tkyi) [reply plugins/pipelines/helper.js]: added

**Comment by tkyi** [comment]
@jithin1987 Updated regex to strip `./` from path

**Pull request merged**